### ISSUE TITLE
"extended_capitalisation_policy = pascal" not working for function names in SQLFluff 3.4.2

### ISSUE THREAD

**Issue by dubf** [sqlfluff/sqlfluff#7150]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The `extended_capitalisation_policy = pascal` setting for function capitalisation does not work as expected. Functions are still flagged as violations even when configured to allow PascalCase.

### Additional Context
The `upper` and `lower` policies work correctly. Only the `pascal` policy seems to be ignored.
This affects readability when trying to distinguish between function names (PascalCase) and table/column names (snake_case).

### Expected Behaviour

With the configuration below, functions like `LastInsertRowId()` and `StrfTime()` should be accepted without linting errors when using PascalCase.

### Observed Behaviour

SQLFluff still flags PascalCase functions as violations and suggests converting them to UPPERCASE, regardless of the pascal policy setting.

### How to reproduce

Use the configuration shown below

Write any function using PascalCase:

### Example SQL
```
SELECT LastInsertRowId();
SELECT StrfTime('%Y-%m-%d', 'now');
```

### Dialect

sqlite

### Version

3.4.2

### Configuration

```
[sqlfluff]
dialect = sqlite

[rules:capitalisation.functions]
extended_capitalisation_policy = pascal
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
In your configuration the section name should be `[sqlfluff:rules:capitalisation.functions]` instead of `[rules:capitalisation.functions]`.
